tools/update_centreon_storage_logs.php

```diff
@@ -0,0 +1,387 @@
+}
+
+$dbUser = 'root';
+
+$insertQuery = <<<'QUERY'
+LOAD DATA INFILE '{{DATA_FILE}}'
+INTO TABLE logs 
+FIELDS TERMINATED BY ','
+LINES TERMINATED BY '\n'
+(ctime, host_id, host_name, instance_name, issue_id, msg_type, notification_cmd, 
+notification_contact, output, retry, service_description, service_id, status, type)
+QUERY;
+
+// We create a loc file to avoid to start this script one more time
+$fileInfos = pathinfo(__FILE__);
+$lockFileName = $fileInfos['filename'] . '.loc';
+if (file_exists($lockFileName)) {
+    mySysLog('Process ' . __FILE__ . ' already running');
+    exit();
+}
+touch($lockFileName);
+
+$currentStep = 1;
+try {
+    // We check if the database connection port is set, otherwise we request it.
+    if (defined('port')) {
+        $dbPort = port;
+    } else {
+        $port = askQuestion("Please enter the port of the Centreon database: ");
+        if (! is_numeric($port)) {
+            throw new Exception("The port given ($port) is not valid");
+        }
+        $dbPort = (int) $port;
+    }
+
+    // We check if the database connection host is set, otherwise we request it.
+    if (defined('hostCentstorage')) {
+        $dbHost = hostCentstorage;
+    } else {
+        $dbHost = askQuestion("Please enter the host or IP address of the Centreon database: ");
+    }
+
+    if (! isset($dbPassword)) {
+        // We need a root access
+        $dbPassword = askQuestion("Please enter the root password of database: ", true);
+    }
+
+    try {
+        $dsn = sprintf(
+            "mysql:dbname=centreon_storage;host=%s;port=%d",
+            $dbHost,
+            $dbPort
+        );
+        $db = new \PDO($dsn, $dbUser, $dbPassword);
+        $db->setAttribute(\PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, false);
+    } catch (\PDOException $ex) {
+        throw new Exception("Can not connect to the database");
+    }
+
+    /*
+     * First we check if this update is necessary by checking if the log_id
+     * column is present
+     */
+    $result = $db->query(
+        "SELECT COUNT(*) AS is_present
+        FROM information_schema.COLUMNS
+        WHERE TABLE_SCHEMA = 'centreon_storage'
+        AND TABLE_NAME = 'logs' AND COLUMN_NAME = 'log_id'"
+    );
+    $shouldBeContinue = ($result->fetchAll(\PDO::FETCH_ASSOC))[0]['is_present'] == '1';
+
+    if (! $shouldBeContinue) {
+        throw new Exception("Update already done");
+    }
+
+    // We will process partitions that are not empty
+    $partitions = getNotEmptyPartitions($db);
+
+    if (! isset($shouldDeleteOldData)) {
+        // We ask to user if he want to keep old data otherwise we will delete the old logs table
+        $explanation = <<<TEXT
+Before starting, we inform you that we will create a new centreon_storage.logs table and rename the older.
+Next, we will copy data from the older table to new one\n\n
+TEXT;
+        printf($explanation);
+
+        $shouldDeleteOldData =
+            !askYesOrNoQuestion("Do you want to keep the old logs table ?");
```

Comment:
To be consistent, shouldn't you add a space after the "!" ?

Reply:
And remove the space before the "?"
https://en.wikipedia.org/wiki/Question_mark#Stylistic_variants
> in the English language orthography no space is allowed in front of the question mark